### PR TITLE
added search result path to the clipboard button

### PR DESCRIPTION
**Pull request by kazuihitoshi**
This is a pull request related to #2646.
I have reviewed and included patterns other than smb paths.
I modified it to a simple button addition that copies the search result path to the clipboard, as a result.
It took a bit of a detour, but I figured that making the path accessible from the client PC should be done in the system configuration.
For files with server-local paths, I assume that they will be converted in the path map.
This is my first patch, please consider merging.
Thanks.

### CODE REVIEW

File: src/main/webapp/WEB-INF/view/searchResults.jsp [hunk 1]
```diff
@@ -1,5 +1,6 @@
 <%@page pageEncoding="UTF-8" contentType="text/html; charset=UTF-8"%>
 <%-- query matched some document --%>
+<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js"></script>
```
Comment (marevol) [inline src/main/webapp/WEB-INF/view/searchResults.jsp]: Fess cannot use files in CDN.
There are js files in src/main/webapp/js.

File: src/main/webapp/WEB-INF/view/searchResults.jsp [hunk 2]
```diff
@@ -38,10 +39,11 @@
 	<ol class="list-unstyled col-md-8">
 		<c:forEach var="doc" varStatus="s" items="${documentItems}">
 			<li id="result${s.index}">
-				<h3 class="title text-truncate">
+				<h3 class="title text-truncate" style="margin-bottom:0;">
```
Comment (marevol) [inline src/main/webapp/WEB-INF/view/searchResults.jsp]: It's better not to use a style attribute.

File: src/main/webapp/WEB-INF/view/searchResults.jsp [hunk 3]
```diff
@@ -226,4 +233,18 @@
 			</c:if>
 		</ul>
 	</nav>
-</div>
+	<script>
```
Comment (marevol) [inline src/main/webapp/WEB-INF/view/searchResults.jsp]: please use search.js.

File: src/main/webapp/WEB-INF/view/searchResults.jsp [hunk 4]
```diff
@@ -38,10 +39,11 @@
 	<ol class="list-unstyled col-md-8">
 		<c:forEach var="doc" varStatus="s" items="${documentItems}">
 			<li id="result${s.index}">
-				<h3 class="title text-truncate">
+				<h3 class="title text-truncate" style="margin-bottom:0;">
 					<a class="link" href="${doc.url_link}" data-uri="${doc.url_link}"
 						data-id="${doc.doc_id}" data-order="${s.index}">${doc.content_title}</a>
-				</h3>
+				</h3>									
+				<p>${doc.url}</p>
```
Comment (marevol) [inline src/main/webapp/WEB-INF/view/searchResults.jsp]: please remove this p tag.

File: src/main/webapp/WEB-INF/view/searchResults.jsp [hunk 5]
```diff
@@ -55,24 +57,29 @@
...
-								href="/search?q=${f:u(q)}&ex_q=${f:u(queryEntry.value)}&sdh=${f:u(fe:sdh(doc.similar_docs_hash))}${fe:facetQuery()}${fe:geoQuery()}">
-								<la:message key="labels.search_result_similar"
-											arg0="${fe:formatFileSize(doc.similar_docs_count-1)}" />
-							</la:link>
-						</small>
-					</c:if>
+				<div class="site">
+					<div class="title-container" style="overflow:unset;">
+						<cite>${f:h(doc.site_path)}</cite>
+						<i class="clipboard-copy fa fa-copy" data-clipboard-text="${doc.url_link}"></i>
```
Comment (marevol) [inline src/main/webapp/WEB-INF/view/searchResults.jsp:61]: Please add only this line.
Other changes are difficult at this time.

**Comment by kazuihitoshi** [comment]
Thanks for pointing that out.
I will correct it, please give me a few minutes.

File: src/main/webapp/WEB-INF/view/searchResults.jsp (hunk 1 above)
Comment (kazuihitoshi) [reply src/main/webapp/WEB-INF/view/searchResults.jsp]: I did not know that the use of a CDN was not allowed.
Thanks for letting me know.
I have corrected it.

File: src/main/webapp/WEB-INF/view/searchResults.jsp (hunk 2 above)
Comment (kazuihitoshi) [reply src/main/webapp/WEB-INF/view/searchResults.jsp]: It was an unnecessary STYLE, so we removed it.
Sorry for the unnecessary code.

File: src/main/webapp/WEB-INF/view/searchResults.jsp (hunk 3 above)
Comment (kazuihitoshi) [reply src/main/webapp/WEB-INF/view/searchResults.jsp]: moved code to search.js.

File: src/main/webapp/WEB-INF/view/searchResults.jsp (hunk 4 above)
Comment (kazuihitoshi) [reply src/main/webapp/WEB-INF/view/searchResults.jsp]: It was an unnecessary code. We have deleted it.
My apologies for the residual debugging code.

File: src/main/webapp/WEB-INF/view/searchResults.jsp (hunk 5 above)
Comment (kazuihitoshi) [reply src/main/webapp/WEB-INF/view/searchResults.jsp:61]: I apologize for my various inadequacies.
Please consider again.

**Comment by kazuihitoshi** [comment]
The corrections you indicated have been completed.
We are very sorry for the inconvenience, but we would appreciate it if you would consider it again.
Thank you.

File: src/main/webapp/WEB-INF/view/searchResults.jsp (hunk 5 above)
Comment (marevol) [reply src/main/webapp/WEB-INF/view/searchResults.jsp:61]: It's difficult to apply this change... 
- UI is changed on some devices
- style attribute needs to be removed

File: src/main/webapp/WEB-INF/view/searchResults.jsp [hunk 6]
```diff
@@ -1,5 +1,6 @@
 <%@page pageEncoding="UTF-8" contentType="text/html; charset=UTF-8"%>
 <%-- query matched some document --%>
+<script type="text/javascript" src="${fe:url('/js/clipboard.min.js')}"></script>
```
Comment (marevol) [inline src/main/webapp/WEB-INF/view/searchResults.jsp]: Please move this line to the bottom of search.jsp.

File: src/main/webapp/WEB-INF/view/searchResults.jsp [hunk 7]
```diff
@@ -41,7 +42,7 @@
 				<h3 class="title text-truncate">
 					<a class="link" href="${doc.url_link}" data-uri="${doc.url_link}"
 						data-id="${doc.doc_id}" data-order="${s.index}">${doc.content_title}</a>
-				</h3>
+				</h3>
```
Comment (marevol) [inline src/main/webapp/WEB-INF/view/searchResults.jsp]: Please remove this change.

File: src/main/webapp/js/search.js [hunk 8]
```diff
@@ -5,7 +5,6 @@ $(function() {
       $searchButton = $("#searchButton"),
       contextPath = $("#contextPath").val(),
       loadImage;
-
```
Comment (marevol) [inline src/main/webapp/js/search.js]: Please remove this change.

File: src/main/webapp/WEB-INF/view/searchResults.jsp (hunk 6 above)
Comment (kazuihitoshi) [reply src/main/webapp/WEB-INF/view/searchResults.jsp]: Thank you for pointing this out.
I have moved it to the bottom of search.jsp.

File: src/main/webapp/WEB-INF/view/searchResults.jsp (hunk 7 above)
Comment (kazuihitoshi) [reply src/main/webapp/WEB-INF/view/searchResults.jsp]: Sorry for the unnecessary correction.
I removed it.

File: src/main/webapp/js/search.js (hunk 8 above)
Comment (kazuihitoshi) [reply src/main/webapp/js/search.js]: I removed it.
Sorry for the unnecessary layers of modifications.

**Comment by kazuihitoshi** [comment]
Thanks for pointing that out.
I have corrected it and it would be helpful if you could check again.
I also double checked and changed the search.js to lf since it was crlf.

File: src/main/webapp/WEB-INF/view/searchResults.jsp (hunk 5 above)
Comment (kazuihitoshi) [reply src/main/webapp/WEB-INF/view/searchResults.jsp:61]: I have Windows 10 21H2(64bit) Chrome 100.0.4896.127(64bit), Firefox 099.0.1(64bit),Internet Explore 11 and it appears to be working fine.

I looked at the Chrome Developer tool with the SmartPhone screen size and could not find any particular problem.

The only thing I can think of is that the clipboard copy is positioned on the right side of the image below.
image

Comment (kazuihitoshi) [reply src/main/webapp/WEB-INF/view/searchResults.jsp:61]: I found the clipboard copy icon in Internet Explore does not work...
I try to fix it. My apologies.

Comment (kazuihitoshi) [reply src/main/webapp/WEB-INF/view/searchResults.jsp:61]: <ul>
  <li>UI is changed on some devices</li>
I was using syntax in search.js that is not supported by some browsers, such as const.
I rewrote it as var, etc. and confirmed that it works correctly in Internet Explorer.
<li>style attribute needs to be removed</li>
I have removed the style addition.
</ul>

Thank you very much for your kind response.
Please review it again.

Thanks

Comment (marevol) [reply src/main/webapp/WEB-INF/view/searchResults.jsp:61]: Thank you for fixing it.
Hmm, I'll consider UI with a clipboard in 14.2 release...

Comment (kazuihitoshi) [reply src/main/webapp/WEB-INF/view/searchResults.jsp:61]: I would appreciate it very much.
Thank you very much for your kind attention to this request.

**Comment by marevol** [comment]
Merged with the working branch.

**Pull request merged**